packages/design-preview/src/components/index.tsx

```diff
@@ -1,17 +1,40 @@
+import { useEffect, useState } from '@wordpress/element';
 import Sidebar from './sidebar';
-import type { StyleVariation } from '@automattic/design-picker';
+import SitePreview from './site-preview';
+import type { StyleVariation } from '@automattic/design-picker/src/types';
 import './style.scss';
 
 interface PreviewProps {
+	previewUrl: string;
 	title?: string;
 	description?: string;
 	variations?: StyleVariation[];
 }
 
-const Preview: React.FC< PreviewProps > = ( { title, description, variations = [] } ) => {
+const Preview: React.FC< PreviewProps > = ( {
+	previewUrl,
+	title,
+	description,
+	variations = [],
+} ) => {
+	const [ activeVariation, setActiveVariation ] = useState< StyleVariation | undefined >();
+
+	useEffect( () => {
+		if ( variations.length > 0 && ! activeVariation ) {
+			setActiveVariation( variations[ 0 ] );
+		}
+	}, [ variations, activeVariation ] );
```

Comment:
Is it possible to remove `activeVariation` from dependencies? Otherwise, this effect will be trigged whenever we change the `activeVariation` and do nothing. Another way is to use `activeIndex` with the default value 0, and then we don't need this effect. What do you think?

Reply:
That makes sense, let me follow this up with a PR with other UI tasks.